programs/clearing_house/src/state/insurance_fund_stake.rs

```diff
@@ -17,3 +27,49 @@ pub struct InsuranceFundStake {
 
     pub cost_basis: i64,
 }
+
+impl InsuranceFundStake {
+    pub fn new(authority: Pubkey, market_index: u64, now: i64) -> Self {
+        InsuranceFundStake {
+            authority,
+            market_index,
+            last_withdraw_request_shares: 0,
+            last_withdraw_request_value: 0,
+            last_withdraw_request_ts: 0,
+            cost_basis: 0,
+            if_base: 0,
+            last_valid_ts: now,
+            if_shares: 0,
+        }
+    }
+
+    pub fn checked_if_shares(&self, spot_market: &SpotMarket) -> ClearingHouseResult<u128> {
+        validate!(
+            self.if_base == spot_market.if_shares_base,
+            ErrorCode::DefaultError,
+            "if stake bases mismatch. user base: {} market base {}",
+            self.if_base,
+            spot_market.if_shares_base
+        )?;
+
+        Ok(self.if_shares)
+    }
+
+    pub fn unchecked_if_shares(&self) -> u128 {
+        self.if_shares
+    }
+
+    pub fn increase_if_shares(&mut self, delta: u128) -> ClearingHouseResult {
```

Comment:
Does it make sense to have a `checked` increase/decrease variant as well here?

Reply:
like pass in the spot market? yeah probably